src/padlocal/message-parser/message-parser-friendship.ts

```diff
@@ -48,6 +48,10 @@ interface ReceiveXmlSchema {
       content: string;
       scene: string;
       ticket: string;
+      sourceusername?: string;
+      sourcenickname?: string;
+      sharecardusername?: string;
+      sharecardnickname?: string;
```

Comment:
Ok, it seems the `sourceXXX` and `shareXXX` name conversion is coming from the WeChat server data payload.

Reply:
yep